language/tools/move-mv-llvm-compiler/src/stackless/translate.rs

```diff
@@ -396,6 +437,42 @@ impl<'mm, 'up> FunctionContext<'mm, 'up> {
                     _ => todo!(),
                 }
             }
+            Operation::ReadRef => {
+                assert_eq!(src.len(), 1);
+                assert_eq!(dst.len(), 1);
+                let src_idx = src[0];
+                let dst_idx = dst[0];
+                let dst_llty = self.locals[dst_idx].llty;
+                let src_llval = self.locals[src_idx].llval;
+                let dst_llval = self.locals[dst_idx].llval;
+                self.llvm_builder
+                    .load_deref_store(dst_llty, src_llval, dst_llval);
+            }
+            Operation::WriteRef => {
+                // nb: both operands are from the "src" vector.
+                // "src" and "dst" might be the wrong names, maybe
+                // "ops" and "returns", since these operations are all
+                // expressed in stackless bytecode as function calls.
+                assert_eq!(src.len(), 2);
+                assert_eq!(dst.len(), 0);
+                let src_idx = src[1];
+                let dst_idx = src[0];
+                let src_llty = self.locals[src_idx].llty;
+                let src_llval = self.locals[src_idx].llval;
+                let dst_llval = self.locals[dst_idx].llval;
+                self.llvm_builder
+                    .load_store_ref(src_llty, src_llval, dst_llval);
+            }
+            Operation::FreezeRef => {
+                assert_eq!(dst.len(), 1);
+                assert_eq!(src.len(), 1);
+                let src_idx = src[0];
+                let dst_idx = dst[0];
+                let src_llty = self.locals[src_idx].llty;
+                let src_llval = self.locals[src_idx].llval;
+                let dst_llval = self.locals[dst_idx].llval;
+                self.llvm_builder.load_store(src_llty, src_llval, dst_llval);
```

Comment:
I don't think we have to care. There are a bunch of semantics Move puts into the bytecode so that it can be typechecked in the VM but at this stage the typechecking is done and we can trust that it is correct, so there are a bunch of bytecode ops that bear semantics that don't matter at the machine level and we can more-or-less ignore. e.g. freezing - it turns a mutable reference to immutable, but within llvm a pointer is a pointer, so freezing is a no-op; and the Move typechecker has already promised not to mutate the value behind the immutable pointer.

That said, we might be able to apply no-aliasing optimizations to immutable references in the future, but it probably doesn't matter much.

That said, I am not actually thinking yet about how to ensure that cross-program Move calls are correct and secure in a Solana execution environment.